src/main/java/duck/Duck.java

```diff
@@ -0,0 +1,216 @@
+        Optional<LocalDate> optionalDate;
+        try {
+            optionalDate = Optional.ofNullable(Parser.parseDate(input));
+            responses.add("Here are the tasks up to the date: "
+                    + optionalDate.get().format(DateTimeFormatter.ofPattern("MMM d yyyy")));
+        } catch (DuckException e) {
+            optionalDate = Optional.empty();
+            responses.add("Here are the tasks sorted by date");
+            responses.add("You can filter up to a date by using \"due [/at | /by] yyyy-mm-dd\"");
+        }
+
+        String[] statusesByDueDate = this.taskList.getStatusesByDate(optionalDate);
+        responses.addAll(Arrays.asList(statusesByDueDate));
+    }
+
+
+    /**
+     * Gets the list of statuses from TaskList and displays them.
+     *
+     * @param input Input from User.
+     * @throws DuckException If description field is empty.
+     */
+    private void listByFind(String input) throws DuckException {
+        input = input.substring(4);
+
+        String[] statusesByFind = this.taskList.getStatusesByFind(input);
+        responses.add("Here are the matching tasks in your list:");
+        Collections.addAll(responses, statusesByFind);
+
+    }
+
+    /**
+     * Marks the task as done and displays the done task.
+     *
+     * @param input Input from user.
+     * @throws DuckException If index obtained is less than 1 or greater than number of tasks.
+     */
+    private void markTaskAsDone(String input) throws DuckException {
+        int taskNumber = Parser.parseTaskNumber(input);
+        Task task = this.taskList.markDone(taskNumber);
+        responses.add("Nice! I've marked this as done");
+        responses.add("  " + task.getStatus());
+        save();
+    }
+
+    /**
+     * Deletes the task and displays the deleted task.
+     *
+     * @param input Input from user.
+     * @throws DuckException If index obtained is less than 1 or greater than number of tasks.
+     */
+    private void deleteTask(String input) throws DuckException {
+        int taskNumber = Parser.parseTaskNumber(input);
+        Task task = this.taskList.deleteTask(taskNumber);
+        responses.add("Noted. I've removed this task");
+        responses.add("  " + task.getStatus());
+        responses.add(getNumberOfTasks());
+        save();
+    }
+
+    /**
+     * Obtains a task from TaskFactory and adds it to the TaskList.
+     *
+     * @param input Input from user.
+     * @throws DuckException If input is unable to be parsed into any Task.
+     */
+    private void createNewTask(String input) throws DuckException {
+        Task newTask = TaskFactory.createTaskFromInput(input);
+        this.taskList.addTask(newTask);
+        responses.add("Got it. I've added this task");
+        responses.add("  " + newTask.getStatus());
+        responses.add(getNumberOfTasks());
+        save();
+    }
+
+
+    /**
+     * Receives input continuously until "bye" command is given.
+     * Main loop of the bot.
```

Comment:
JavaDocs can be updated to account for the return type of String[] this time

Reply:
Thanks for pointing that out!